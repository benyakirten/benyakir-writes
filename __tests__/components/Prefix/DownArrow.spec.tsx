import * as React from 'react'
import renderer, { act } from 'react-test-renderer'
import 'jest-styled-components'

import { DownArrow } from '@Pre'

describe('DownArrow component', () => {
  it('should render properly', () => {
    expect(() => renderer.create(<DownArrow />)).not.toThrow()
  })

  it('should rotate the arrow based on the open prop', async () => {
    const container = renderer.create(<DownArrow />)
    expect(container.toJSON()).toMatchSnapshot()
    expect(container.toJSON()).toHaveStyleRule('transform', 'rotate(0deg)')

    renderer.act(() => {
      container.update(<DownArrow open={true} />)
    })
    expect(container.toJSON()).toHaveStyleRule('transform', 'rotate(180deg)')
  })

  it('should render a span with the same tabIndex as the prop', () => {
    const container = renderer.create(<DownArrow />)
    expect(container.toJSON()).toMatchSnapshot()
    expect((container.toJSON() as any).props.tabIndex).toEqual(-1)

    renderer.act(() => {
      container.update(<DownArrow tabIndex={0} />)
    })
    expect((container.toJSON() as any).props.tabIndex).toEqual(0)
  })

  it('should render a span with the same onClick as the prop', () => {
    const container = renderer.create(<DownArrow />)
    expect(container.toJSON()).toMatchSnapshot()
    expect((container.toJSON() as any).props.onClick).toBeUndefined()

    const testClick = () => 1 + 1
    act(() => {
      container.update(<DownArrow onClick={testClick} />)
    })
    expect((container.toJSON() as any).props.onClick).toEqual(testClick)
  })
})
