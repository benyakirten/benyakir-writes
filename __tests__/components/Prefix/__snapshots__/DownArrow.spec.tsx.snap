// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownArrow component should render a span with the same onClick as the prop 1`] = `
.c0 {
  position: absolute;
  left: -1.5rem;
  top: 0.4rem;
  outline: none;
  font-size: 2.4rem;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<span
  className="c0"
  data-cy="submenu-open"
  open={false}
  role="button"
  tabIndex={-1}
>
  ↡
</span>
`;

exports[`DownArrow component should render a span with the same tabIndex as the prop 1`] = `
.c0 {
  position: absolute;
  left: -1.5rem;
  top: 0.4rem;
  outline: none;
  font-size: 2.4rem;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<span
  className="c0"
  data-cy="submenu-open"
  open={false}
  role="button"
  tabIndex={-1}
>
  ↡
</span>
`;

exports[`DownArrow component should rotate the arrow based on the open prop 1`] = `
.c0 {
  position: absolute;
  left: -1.5rem;
  top: 0.4rem;
  outline: none;
  font-size: 2.4rem;
  -webkit-transition: all 1s ease;
  transition: all 1s ease;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<span
  className="c0"
  data-cy="submenu-open"
  open={false}
  role="button"
  tabIndex={-1}
>
  ↡
</span>
`;
